hibernate-core/src/main/java/org/hibernate/sql/ast/tree/insert/InsertStatement.java

```diff
@@ -7,60 +7,73 @@
 package org.hibernate.sql.ast.tree.insert;
 
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 
 import org.hibernate.sql.ast.tree.MutationStatement;
+import org.hibernate.sql.ast.tree.cte.CteConsumer;
 import org.hibernate.sql.ast.tree.expression.ColumnReference;
-import org.hibernate.sql.ast.tree.expression.Expression;
 import org.hibernate.sql.ast.tree.from.TableReference;
+import org.hibernate.sql.ast.tree.select.QuerySpec;
+
+import org.jboss.logging.Logger;
 
 /**
  * @author Steve Ebersole
  */
-public class InsertStatement implements MutationStatement {
+public class InsertStatement implements MutationStatement, CteConsumer {
+	private static final Logger log = Logger.getLogger( InsertStatement.class );
+
 	private TableReference targetTable;
 	private List<ColumnReference> targetColumnReferences;
-	private List<Expression> values;
-
-	public InsertStatement(TableReference targetTable) {
-		this.targetTable = targetTable;
-	}
+	private QuerySpec sourceSelectStatement = null;
```

Comment:
We never need to set `null` value for JVM has done that for us already, right?

Reply:
I fixed it.